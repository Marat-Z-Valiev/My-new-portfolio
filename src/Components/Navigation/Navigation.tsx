import React, {useState, useEffect} from "react";
import NavLink from "../NavigationLink/NavigationLink";
import NavStyled from "./NavigationStyled";
import headerLinks from "../../data/headerLinks";

interface NavigationProps {
	link: string;
}

const Navigation: React.FC<NavigationProps> = () => {
	let [isSticky, setStickyNav] = useState(false);

	const handleScroll = () => {
		const header = document.querySelector("header") as HTMLElement;
		const ul = document.querySelector(".menu") as HTMLElement;
		const app = document.getElementById("app") as HTMLElement;
		if (window.pageYOffset > header.offsetTop) {
			app.style.marginTop = `${ul.offsetHeight}px`;
			setStickyNav((isSticky = true));
		} else {
			app.style.marginTop = `0px`;
			setStickyNav((isSticky = false));
		}
	};

	useEffect(() => {
		window.addEventListener("scroll", handleScroll);
	}, []);

	const toggleMenu = () => {
		const links = document.querySelector(".links") as HTMLElement;
		links.classList.toggle("open");
	};

	const closeMenu = () => {
		const hamburgerButton = document.querySelector(
			".hamburger-button"
		) as HTMLButtonElement;
		hamburgerButton.click();
	};

	const closeMenuAfterClick = () => {
		const liElements = document.querySelectorAll(".links .nav-list-item");
		liElements.forEach((element) => {
			element.addEventListener("click", closeMenu);
		});
	};

	useEffect(() => {
		closeMenuAfterClick();
	}, []);

	const stickyClass = isSticky ? "fixed" : "";

	return (
		<NavStyled>
			<header className={stickyClass}>
				<nav>
					<ul className="menu">
						<li className="logo">
							<h1>Marat Valiev</h1>
						</li>
						<li className="toggle">
							<button
								type="button"
								className="hamburger-button"
								onClick={toggleMenu}
							>
								<i className="fas fa-bars fa-3x"></i>
							</button>
						</li>
						<div className="links">
							{headerLinks.map((menuItem, index) => (
								<NavLink
									className="item"
									key={index}
									link={menuItem.link}
									label={menuItem.label}
								/>
							))}
						</div>
					</ul>
				</nav>
			</header>
		</NavStyled>
	);
};

export default Navigation;
