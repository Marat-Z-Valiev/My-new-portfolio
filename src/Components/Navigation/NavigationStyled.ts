import styled from "styled-components";

const NavStyled = styled.div`
	header {
		position: absolute;
		top: 0;
		width: 100%;
		background-color: transparent;
		box-shadow: none;
		padding-top: 15px;
		padding-bottom: 15px;
		transition: 0.2s background-color ease-in-out, 0.2s box-shadow ease-in-out,
			0.2s padding-top ease-in-out, 0.2s padding-bottom ease-in-out;
	}

	.fixed {
		position: fixed;
		z-index: 100;
		box-shadow: 0px 2px 4px -1px rgba(0, 0, 0, 0.2),
			0px 4px 5px 0px rgba(0, 0, 0, 0.14), 0px 1px 10px 0px rgba(0, 0, 0, 0.12);
		background-color: #1175c0;
		padding-top: 0;
		padding-bottom: 0;
	}

	.menu {
		display: flex;
		justify-content: space-around;
		align-items: center;
		height: 60px;
		z-index: 5;
		margin-top: 0;

		.links {
			display: flex;
			width: 500px;
			justify-content: space-around;

			@media (min-width: 320px) and (max-width: 425px) {
				margin-left: -40px;
				flex-direction: column;
				text-align: center;
				padding: 15px 5px;
				transform: translateX(-120%);
				transition: transform 0.5s ease-out;
				background-color: #1175c0;
				width: 100vw;

				li {
					justify-content: space-between;
				}
			}
		}

		.toggle {
			display: none;
		}

		h1 {
			font-size: 2.3rem;
		}

		li {
			list-style: none;
			font-size: 1.3rem;
		}
		a {
			text-decoration: none;
			color: #ffffff;

			&:hover {
				color: #11ee1c;
				text-decoration: underline;
				cursor: pointer;
			}
		}
		a.active {
			color: #11ee1c;
		}

		@media (min-width: 320px) and (max-width: 425px) {
			display: flex;
			flex-direction: row;
			justify-content: space-between;
			align-items: center;
			flex-wrap: wrap;
			height: 100px;

			.open {
				transform: translateX(0);
			}

			.toggle {
				display: block;

				button {
					background: transparent;
					border: none;
					color: #ffffff;
					margin-right: 60px;
				}
			}
		}
	}
`;

export default NavStyled;
