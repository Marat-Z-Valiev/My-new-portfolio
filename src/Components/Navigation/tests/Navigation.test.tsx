import React from 'react';
import { render } from "@testing-library/react";
import Navigation from "../Navigation";

describe("Navigation", () => {
  it("should render correctly", () => {
    render(<Navigation />);
  });
});
