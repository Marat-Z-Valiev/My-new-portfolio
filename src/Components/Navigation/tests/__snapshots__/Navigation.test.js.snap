// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render correctly 1`] = `
<NavigationStyled__NavStyled>
  <header
    className=""
  >
    <nav>
      <ul
        className="menu"
      >
        <li
          className="logo"
        >
          <h1>
            Marat Valiev
          </h1>
        </li>
        <li
          className="toggle"
        >
          <button
            className="hamburger-button"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-bars fa-3x"
            />
          </button>
        </li>
        <div
          className="links"
        >
          <NavigationLink
            className="item"
            key="0"
            label="PROJECTS"
            link="projects"
          />
          <NavigationLink
            className="item"
            key="1"
            label="SKILL SETS"
            link="skillSet"
          />
          <NavigationLink
            className="item"
            key="2"
            label="CONTACT"
            link="contact"
          />
        </div>
      </ul>
    </nav>
  </header>
</NavigationStyled__NavStyled>
`;
